usaspending_api/agency/v2/views/sub_agency.py

```diff
@@ -0,0 +1,138 @@
+from elasticsearch_dsl import A
+from rest_framework.request import Request
+from rest_framework.response import Response
+from typing import Any
+from usaspending_api.agency.v2.views.agency_base import AgencyBase, PaginationMixin
+from fiscalyear import FiscalYear
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.elasticsearch.search_wrappers import TransactionSearch
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.common.query_with_filters import QueryWithFilters
+
+
+class SubAgencyList(PaginationMixin, AgencyBase):
+    """
+    Obtain the list of subagencies and offices based on the provided toptier_code and fiscal year, as well as award type.
+    Can be either by funding agency or awarding agency.
+    Based on transaction data.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/sub_agency.md"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params_to_validate = ["fiscal_year", "agency_type", "award_type"]
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        self.sortable_columns = ["name", "total_obligations", "transaction_count", "new_award_count"]
+        self.default_sort_column = "total_obligations"
+        results = sorted(
+            self.get_sub_agency_list(),
+            key=lambda x: x.get(self.pagination.sort_key),
+            reverse=self.pagination.sort_order == "desc",
+        )
+        page_metadata = get_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        return Response(
+            {
+                "toptier_code": self.toptier_code,
+                "fiscal_year": self.fiscal_year,
+                "page_metadata": page_metadata,
+                "results": results[self.pagination.lower_limit : self.pagination.upper_limit],
+                "messages": self.standard_response_messages,
+            }
+        )
+
+    def format_elasticsearch_results(self, results):
+        response = []
+        buckets = results.aggs.to_dict().get("subtier_agencies", {}).get("buckets", [])
+        for bucket in buckets:
+            response.append(
+                {
+                    "name": bucket.get("key"),
+                    "abbreviation": bucket.get("subagency_info")
+                    .get("hits")
+                    .get("hits")[0]
+                    .get("_source")
+                    .get(f"{self.agency_type}_subtier_agency_abbreviation"),
+                    "total_obligations": bucket.get("total_subagency_obligations").get("value"),
```

Comment:
Similar to your other PR, need to make sure to handle sum of floats since Elasticsearch has proven to not be that great at it. Example response I got:
```
{
    "toptier_code": "097",
    "fiscal_year": 2020,
    "page_metadata": {
        "page": 1,
        "total": 10,
        "limit": 10,
        "next": null,
        "previous": null,
        "hasNext": false,
        "hasPrevious": false
    },
    "results": [
        ...
        {
            "name": "Defense Commissary Agency",
            "abbreviation": "DECA",
            "total_obligations": 630639.4500000001,
            "transaction_count": 15,
            "new_award_count": 0,
            "children": []
        }
    ],
    "messages": []
}
```